src/System.Windows.Forms/src/System/Windows/Forms/AxHost.State.cs

```
@@ -274,23 +271,25 @@ private void InitializeBufferFromStream(Stream ids)
             internal unsafe State? RefreshStorage(IPersistStorage.Interface iPersistStorage)
             {
                 Debug.Assert(_storage is not null, "how can we not have a storage object?");
-                Debug.Assert(_iLockBytes is not null, "how can we have a storage w/o ILockBytes?");
-                if (_storage is null || _iLockBytes is null)
+                Debug.Assert(_lockBytes is not null, "how can we have a storage w/o ILockBytes?");
+                if (_storage is null || _lockBytes is null)
                 {
                     return null;
                 }
 
-                iPersistStorage.Save(_storage, fSameAsLoad: true);
-                _storage->Commit(0);
+                using var storage = _storage.GetInterface();
+                iPersistStorage.Save(storage, fSameAsLoad: true);
+                storage.Value->Commit(0);
```

This returns an `HRESULT`. Does it need to be checked?

---

Yes, these were missed when this was converted. We'll make sure to do another full audit compared to Framework to keep throw scenarios the same. This is a little more complicated than "throw everywhere" as some things had `PreserveSig` and there are numerous places where we swallow all exceptions. While it is a bit of a pain, it's nice to see the logic explicitly now.